module/VuFind/src/VuFind/Auth/ILS.php

```diff
@@ -178,6 +200,55 @@ public function updatePassword($request)
         $result = $this->getCatalog()->changePassword(
             [
                 'patron' => $patron,
+                'username' => $patron['cat_username'],
+                'oldPassword' => $params['oldpwd'],
+                'newPassword' => $params['password'],
+            ]
+        );
+        if (!$result['success']) {
+            throw new AuthException($result['status']);
+        }
+
+        // Update the user and send it back to the caller:
+        $username = $patron[$this->getUsernameField()];
+        $user = $this->getOrCreateUserByUsername($username);
+        $this->authenticator->saveUserCatalogCredentials($user, $patron['cat_username'], $params['password']);
+        return $user;
+    }
+
+    /**
+     * Change/Reset a user's password when they've forgotten.
+     *
+     * @param Request $request Request object containing new account details.
+     *
+     * @return UserEntityInterface Updated user entity.
+     *
+     * @throws PasswordSecurity
+     * @throws AuthException
+     * @throws \Exception
+     */
+    public function newPassword($request)
```

Comment:
I'm not entirely sure (without spending more time digging deeper into the code and the history) why we need to split updatePassword into two methods... but if this is unavoidable, I wonder if `resetPassword` would be a better name than `newPassword` since it's more verb-based. Also, if we make this change here, do we need to make parallel changes to `VuFind\Auth\Database` since it is currently implemented to do password resets using only updatePassword, and some of the other code changes here might conflict with that?

Reply:
I'm pretty sure that updatePassword is able to do everything needed, so this method shouldn't be needed.